File: reporter.py
#!/usr/bin/env python

import os
import glob
import re
import time
import sets
import httplib
from urlparse import urlparse
from urllib import quote

conf = {
  "stats_dir": "c:\\Program Files\\rFactor\\UserData\\LOG\\Results\\", 
  "interval": 10.0,
  "post_url": "http://YOURHOST.com/report.php",
  "season_name": "SEASON NAME"
  }

def post_file(url, filename):
  if(os.path.exists(filename)):
    f = open(filename, 'rb')
    conn = httplib.HTTPConnection(url.netloc)
    headers = {"Content-type": "text/xml"}
    conn.request("POST", url.path+"?season="+quote(conf["season_name"]), f.read(), headers)
    f.close()
    response = conn.getresponse()
    print response.read()
  
def main():
  url = urlparse(conf["post_url"])
  existing_files = sets.Set(glob.glob(conf["stats_dir"] + "*SR.xml"))
  print existing_files
  while(True):
    files = sets.Set(glob.glob(conf["stats_dir"] + "*SR.xml"))
    print "."
    new_files = files.difference(existing_files)
    #print new_files
    if(len(new_files) > 0):
      for f in new_files:
        print f
        post_file(url, f)
    existing_files = files
    time.sleep(conf["interval"])
  

if __name__ == "__main__":
  main()

File: reporter.pl
#!/usr/bin/env perl

use LWP::UserAgent;
use HTTP::Request::Common;
use URI::Escape ('uri_escape');

#$stats_dir = "c:\\Program Files\\rFactor\\UserData\\LOG\\Results\\"; 
$stats_dir = "/tmp/"; 
$interval = 10.0;
#$post_url = "http://YOURHOST.com/report.php";
$post_url = "http://localhost/report.php";
$season_name = "SEASON NAME";

sub post_file {
  $filename = $_[0];
  open(F, $filename) || return;
  @raw_data=<F>;
  close(F);
  print "$post_url?season=".uri_escape($season_name)."\n";
  $req = HTTP::Request->new(POST => $post_url."?season=".uri_escape($season_name));
  $req->content(join("", @raw_data));
  $ua = LWP::UserAgent->new;
  $res = $ua->request($req);
  if ($res->is_success) {
    print $res->content;
  } else {
    print $res->status_line, "\n";
    print $res;
  }
}


@existing_files = <$stats_dir*SR.xml>;
print "existing files:\n";
print join("\n", @existing_files);
print "\n";

while(true) {
  @new_files = <$stats_dir*SR.xml>;
  %count = ();
  @intersection = @difference = ();
  foreach $element (@existing_files, @new_files) { $count{$element}++ }
  foreach $element (keys %count) {
    push @{ $count{$element} > 1 ? \@intersection : \@difference }, $element;
  }
  if (scalar(@difference) > 0) {
    foreach $file (@difference) {
      print $file . "\n";
      post_file($file);
    }
  }
  print ".\n";
  @existing_files = @new_files;
  sleep $interval;
}